homeassistant/components/here_travel_time/sensor.py

```diff
@@ -216,236 +225,64 @@ def _are_valid_client_credentials(here_client: herepy.RoutingApi) -> bool:
     return True
 
 
-class HERETravelTimeSensor(SensorEntity):
+class HERETravelTimeSensor(SensorEntity, CoordinatorEntity):
     """Representation of a HERE travel time sensor."""
 
     def __init__(
         self,
         name: str,
-        origin: str,
-        destination: str,
-        origin_entity_id: str,
-        destination_entity_id: str,
-        here_data: HERETravelTimeData,
+        traffic_mode: bool,
+        coordinator: HereTravelTimeDataUpdateCoordinator,
     ) -> None:
         """Initialize the sensor."""
-        self._name = name
-        self._origin_entity_id = origin_entity_id
-        self._destination_entity_id = destination_entity_id
-        self._here_data = here_data
-        self._unit_of_measurement = TIME_MINUTES
-        self._attrs = {
-            ATTR_UNIT_SYSTEM: self._here_data.units,
-            ATTR_MODE: self._here_data.travel_mode,
-            ATTR_TRAFFIC_MODE: self._here_data.traffic_mode,
-        }
-        if self._origin_entity_id is None:
-            self._here_data.origin = origin
-
-        if self._destination_entity_id is None:
-            self._here_data.destination = destination
-
-    async def async_added_to_hass(self) -> None:
-        """Delay the sensor update to avoid entity not found warnings."""
-
-        @callback
-        def delayed_sensor_update(event):
-            """Update sensor after Home Assistant started."""
-            self.async_schedule_update_ha_state(True)
-
-        self.hass.bus.async_listen_once(
-            EVENT_HOMEASSISTANT_START, delayed_sensor_update
-        )
+        super().__init__(coordinator)
+        self._traffic_mode = traffic_mode
+        self._attr_native_unit_of_measurement = TIME_MINUTES
+        self._attr_name = name
 
     @property
     def native_value(self) -> str | None:
         """Return the state of the sensor."""
-        if self._here_data.traffic_mode and self._here_data.traffic_time is not None:
-            return str(round(self._here_data.traffic_time / 60))
-        if self._here_data.base_time is not None:
-            return str(round(self._here_data.base_time / 60))
-
+        if self.coordinator.data is not None:
+            if self._traffic_mode:
+                if (
+                    time := self.coordinator.data.get(ATTR_DURATION_IN_TRAFFIC)
+                ) is not None:
+                    return str(round(time))
+            if (time := self.coordinator.data.get(ATTR_DURATION)) is not None:
+                return str(round(time))
```

Comment:
These fields are never absent or `None`, so this deserves simplification, something like:

```suggestion
            return str(round(self.coordinator.data.get(ATTR_DURATION_IN_TRAFFIC if self._traffic_mode else ATTR_DURATION)))
```
(though the syntax/style/wrapping may be off)